helper/resource/testing.go

```diff
@@ -824,6 +831,17 @@ func Test(t testing.T, c TestCase) {
 		}
 	}(helper)
 
+	// Run the TerraformVersionChecks if we have it.
+	// This is done after creating the helper because a working directory is required
+	// to retrieve the Terraform version.
+	if c.TerraformVersionChecks != nil {
+		logging.HelperResourceDebug(ctx, "Calling TestCase runTFVersionChecks")
```

Comment:
Nit for clearer developer troubleshooting:

```suggestion
		logging.HelperResourceDebug(ctx, "Calling TestCase Terraform version checks")
```